src/WordPress/Hooks.php

```diff
@@ -242,4 +246,63 @@ public function http2ServerPushInit()
     {
         HTTP2ServerPush::init();
     }
+
+    /*********************************** Admin Bar Button ***********************************/
+
+    public function adminBarCachePurgeButtonJS() { 
+        ?>
+        <script type="text/javascript" >
+             jQuery("li#wp-admin-bar-cloudflare-purge-cache .ab-item").on( "click", function() {
+                var data = {
+                  'action': 'cloudflare_admin_bar_purge_cache_button',
+                };
+
+                jQuery.post(ajaxurl, data, function(response) {
+                   alert( response );
+                });
+              });
+          </script> <?php
+        }
+
+    public function adminBarCachePurgeButtonCallback() {
+        $isPurged = $this->purgeCacheEverything(true);
+
+        $responseMessage = __("Cloudflare cache was succsesfully purged");
+        if (!$isPurged) {
+            $responseMessage = __("There was an error purging Cloudflare cache");
+        }        
+        echo $responseMessage;
+
+        // This is required to terminate immediately and return a response message
+        wp_die();
+    } 
+
+    public function addAdminBarMenuButton($wpAdminBar) {
+        $args = array(
+            'id' => 'cloudflare-purge-cache',
+            'title' => 'Cloudflare Purge Cache',
+            'href' => '#'
+        );
+
+        $wpAdminBar->add_node($args);
+    }
+
+    public function shouldShowAdminBarMenuButton() {
+        // TODO add showadminbar cache to not query everytime the page changes
```

Comment:
All of the TODO's are still TODO's. I wanted to push and show the way I'm taking before going further in depth. (read the PR description)